### PR TITLE
Progress on FE task

### PR DESCRIPTION
**Pull request by rosscournoyer602**
A couple notes:

- I'd like to add more features such as 'likes' on photos
- I also hosted a back-end for the content. I decided to host it because it assumes some DB config and I didn't want to bother you with that on FE focused task. It is not yet documented, but I can provide this if you are interested.

### CODE REVIEW

**Comment by flyingant** [comment]
Hi @rosscournoyer602 

This is MaYi from Wiredcraft and reviewing your code.

I noticed you had lots of the code are using the `document` object to get the form value. 

for example
https://github.com/Wiredcraft/test-frontend/pull/36/files#diff-3bba4ff499553627d27a92ccc3dce554b166b431e342a1f82d919b11c6ce0107R23

Do think if you could improve it to make it more react-ish?

Thanks

File: package.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+  "name": "react-simple",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
```
Comment (johncalvinroberts) [inline package.json:6]: No build script?

File: src/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  // when location changes check for query param and execute the search
+  useEffect(() => {
+    const { search } = props.location
+    const { q } = queryString.parse(search)
+
+    if (q) {
+      searchPhotos(q).then(res => {
+        updateImages(res.data)
+      })
+    }
```
Comment (johncalvinroberts) [inline src/Home.jsx]: I've found that `useEffect` is a super-powerful footgun, especially when it comes to data fetching. 

* Did you run into any issues with this data fetching pattern you used here on `Home.jsx`?
* Can you refactor this to use async/await syntax?

File: src/components/Header.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      const id = localStorage.getItem('user')
+      const token = localStorage.getItem('token')
+      if (id && token) {
+        getUser(id, token)
+      }
+    }
+  }, [])
+
+	return (
+		<div className="header">
```
Comment (johncalvinroberts) [inline src/components/Header.jsx]: I've noticed some weird formatting. Might want to run `prettier` on the codebase.

File: src/components/profile/Edit.jsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+			const res = reader.result
+			setPageData(prevState => ({
+				...prevState,
+				fileSrc: res
+			}))
+		}
+		if (acceptedFiles[0]) {
+			reader.readAsDataURL(acceptedFiles[0])	
+		}
+	}, [])
```
Comment (johncalvinroberts) [inline src/components/profile/Edit.jsx]: I think you're missing some deps for the `useCallback` hook.

File: src/components/profile/Edit.jsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+  // react-dropzone helpers
+	const {getRootProps, getInputProps } = useDropzone({onDrop})
```
Comment (johncalvinroberts) [inline src/components/profile/Edit.jsx:32]:
```suggestion
	const { getRootProps, getInputProps } = useDropzone({onDrop})
```

Comment (johncalvinroberts) [reply src/components/profile/Edit.jsx:32]: Great choice of library btw, `react-dropzone` is a good one.

File: src/components/profile/Login.jsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+      <div className="form-field">
+        <label htmlFor="password">
+          Password
+        </label>
+        <input id="password" name="password" className={`form-field__input ${pwErr ? 'form-field__input--error' : ''}`} type="text"></input>
+      </div>
+      <button type="button" onClick={validateForm}>Login</button>
+    </div>
+  )
+}
```
Comment (johncalvinroberts) [inline src/components/profile/Login.jsx]: Could you refactor this to be a bit more "semantic html" friendly? I see you used a different pattern on the login vs. register components.

File: src/components/tests/image.test.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { shallow } from '../enzyme'
+import Image from '../Image'
+
+describe('Image test', () => {
+  
+  it('image renders', () => {
+    shallow(<Image src="https://picsum.photos/240/392?random=286" />)
```
Comment (johncalvinroberts) [inline src/components/tests/image.test.js]: No assertion?

File: webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+module.exports = () => {
+
+  const env = dotenv.config().parsed
+  let envKeys = []
+  if (env) {
+    envKeys = Object.keys(env).reduce((prev, next) => {
+      prev[`process.env.${next}`] = JSON.stringify(env[next])
+      return prev
+    }, {})
+  }
```
Comment (johncalvinroberts) [inline webpack.config.js]: You declare `envKeys` as an array on line 8, then assign it as an object on line 10. No bug but it doesn't help with readability.

**Review by johncalvinroberts — COMMENTED** [review]
Hi Ross. John from Wiredcraft here. Thanks for the PR! I left a few comments, feel free to improve as you see fit.

**Comment by rosscournoyer602** [comment]
> Hi @rosscournoyer602
> 
> This is MaYi from Wiredcraft and reviewing your code.
> 
> I noticed you had lots of the code are using the `document` object to get the form value.
> 
> for example
> https://github.com/Wiredcraft/test-frontend/pull/36/files#diff-3bba4ff499553627d27a92ccc3dce554b166b431e342a1f82d919b11c6ce0107R23
> 
> Do think if you could improve it to make it more react-ish?
> 
> Thanks

Hi MaYi, thanks for the suggestion. I used refs instead.

btw, is there a better more react style way of doing this? 

`const imageTags = Array.from(document.getElementsByClassName('image'));`

I want to get all my image elements to pass them to intersection observer. Just curious.

File: src/Home.jsx (hunk 2 above)
Comment (rosscournoyer602) [reply src/Home.jsx]: I agree. You can get yourself into trouble with too many component re-renders and unexpected side-effects. In this case, it looks like the component does just render once when it mounts and then whenever search changes. Are there any other issues I might look out for here?  I did go ahead and use async/await though.

File: src/components/Header.jsx (hunk 3 above)
Comment (rosscournoyer602) [reply src/components/Header.jsx]: Done!

File: src/components/profile/Edit.jsx (hunk 4 above)
Comment (rosscournoyer602) [reply src/components/profile/Edit.jsx]: I honestly get a little confused with this sometimes, and often depend on a linter or create-react-app to warn me about it. I added what i think should be the correct dependencies here. Please correct if I still have it wrong.

File: src/components/profile/Login.jsx (hunk 6 above)
Comment (rosscournoyer602) [reply src/components/profile/Login.jsx]: Made them both forms

File: src/components/tests/image.test.js (hunk 7 above)
Comment (rosscournoyer602) [reply src/components/tests/image.test.js]: I need to learn more about jest and enzyme, honestly. I hope to learn about this by example = )

File: webpack.config.js (hunk 8 above)
Comment (rosscournoyer602) [reply webpack.config.js]: Yes thanks. I made sure its consistently an array.

File: package.json (hunk 1 above)
Comment (rosscournoyer602) [reply package.json:6]: I tend to use create-react-app and I know what their build script does, not sure how to set that up on my own though. I think I have just the most basic webpack setup here but would like to know how to customize and expand upon it.

**Pull request closed without merging**